anjali/sb-pt-cont/8.js

```diff
@@ -0,0 +1,131 @@
+// https://levelbuilder-studio.code.org/s/sb-pt-cont/lessons/1/levels/8
+/*
+Add at least two other sprites to your Story Board. Give each sprite a unique costume and location!
+*/
+if (!validationProps.successCriteria) {
+  validationProps.successCriteria = {
+    threeSprites: false,
+    differentCostumes: false,
+    differentLocations: false
+  };
+}
+
+if (!validationProps.previous) {
+  validationProps.previous = {
+  	waitTime: 120
+  };
+}
+
+// Helper variables
+var spriteIds = getSpriteIdsInUse();
+
+// Check for at least 3 sprites
+validationProps.successCriteria.threeSprites = spriteIds.length>=3;
+
+
+if (World.frameCount == 1) {
+  // Check sprite locations
+  var uniqueStartingSpriteLocations = [];
+  for (var i=0; i<spriteIds.length; i++) {
+    var coords = [getProp({id: spriteIds[i]}, "x"), getProp({id: spriteIds[i]}, "y")];
+    var noDuplicateCoords = true;
+    for (var j=0; j<uniqueStartingSpriteLocations.length; j++) {
+      if ((coords[0] == uniqueStartingSpriteLocations[j][0]) &&
+          (coords[1] == uniqueStartingSpriteLocations[j][1])) {
+        noDuplicateCoords = false;
+        break;
+      }
+    }
+    if (!noDuplicateCoords) {
+      break;
+    } else {
+      uniqueStartingSpriteLocations.push(coords);
+    }
+  }
+  if (spriteIds.length == uniqueStartingSpriteLocations.length) {
+    validationProps.successCriteria.differentLocations = true;
+  }
+
+  // Check sprite costumes
+  var uniqueStartingSpriteCostumes = [];
+  for (var i=0; i<spriteIds.length; i++) {
+    var costume = getProp({id: spriteIds[i]}, "costume");
+    var noDuplicateCostumes = true;
+    for (var j=0; j<uniqueStartingSpriteCostumes.length; j++) {
+      if (costume == uniqueStartingSpriteCostumes[j]) {
+        noDuplicateCostumes = false;
+        break;
+      }
+    }
+    if (!noDuplicateCostumes) {
+      break;
+    } else {
+      uniqueStartingSpriteCostumes.push(costume);
+    }
+  }
+  if (spriteIds.length == uniqueStartingSpriteCostumes.length) {
+    validationProps.successCriteria.differentCostumes = true;
+  }
+}
+
+// Check if student is interacting with the program
+// version 1
+for (var spriteId in spriteIds) {
+  if (getProp({id: spriteId}, "timeout")>0) {
+    validationProps.waitTime = World.frameCount + 60;
+    break;
+  } else {
+    validationProps.waitTime = validationProps.previous.waitTime;
+  }
+}
+
+/*
+// version 2
```

Comment:
here and elsewhere, delete unused code

Reply:
I was leaving this in because I still have to run by Mike which version of the delayed success bar he likes better.